### PR TITLE
devel: update local containerised development

### PR DESCRIPTION
**Pull request by jkozol**
Update multiple aspects of our docker development stack. The frontend now has a devel specific webpack that allows development against a local server/api. The docker-composer.yaml also includes both versioning and networking updates. Overall, the major change is to migrate our proxy access into the crc webpack proxy.

### CODE REVIEW

**Review by ochosi — COMMENTED** [review]
I tried to quickly test-drive this locally. The setup seems to have gone fine as far as I can tell from the console output, but I ran into an Akamai error:
```
Access Denied
See https://<redacted>/akamai_pre_prod_lockdown
```

**Comment by kingsleyzissou** [comment]
I'm trying to run this to verify, just having issues reaching the backend. I'm getting a proxy error

File: config/devel.webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const { config: webpackConfig, plugins } = config({
+  rootFolder: resolve(__dirname, '../'),
+  debug: true,
+  useFileHash: false,
+  sassPrefix: '.imageBuilder',
+  deployment: 'beta/apps',
+  appUrl: '/preview/insights/image-builder',
+  env: 'prod-beta',
+  useProxy: true,
+  useAgent: false,
```
Comment (lucasgarfield) [inline config/devel.webpack.config.js:14]:
```suggestion
  useAgent: true,
  ```

File: config/devel.webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const config = require('@redhat-cloud-services/frontend-components-config');
+
+const { config: webpackConfig, plugins } = config({
+  rootFolder: resolve(__dirname, '../'),
+  debug: true,
+  useFileHash: false,
+  sassPrefix: '.imageBuilder',
+  deployment: 'beta/apps',
+  appUrl: '/preview/insights/image-builder',
+  env: 'prod-beta',
```
[the remark this answers is not in the document]
Comment (lucasgarfield) [reply config/devel.webpack.config.js:12]: I noticed that in the readme, you changed the URL to `stage.foo.redhat.com`... we should set the environment to `stage` in that case, I think.

File: config/devel.webpack.config.js (hunk 1 above)
Comment (lucasgarfield) [reply config/devel.webpack.config.js:14]: If we want to use the `stage` environment, we need to also set `useAgent` to `true`.

File: devel/README.md [hunk 3]
```diff
@@ -61,13 +61,13 @@ The config variables for the Image Builder backend can be found [here](https://g
 ## Run
 
 ```bash
-docker-compose up --build
+docker compose up
```
Comment (lucasgarfield) [inline devel/README.md:64]:
```suggestion
docker compose build
docker compose up
```

File: config/devel.webpack.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkozol) [reply config/devel.webpack.config.js:12]: Oops. Missed this while playing around getting it to run

**Comment by jkozol** [comment]
I don't know what I did to get this working haha. It was working on my mac but now on my fedora machine it does not work. It will run but won't connect to the api now. @lucasgarfield are you able to run it against the local backend?

File: devel/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkozol) [reply devel/README.md:64]: I will add a section stating to make sure you have an updated version of docker

**Comment by lucasgarfield** [comment]
> I don't know what I did to get this working haha. It was working on my mac but now on my fedora machine it does not work. It will run but won't connect to the api now. @lucasgarfield are you able to run it against the local backend?

Hm, I seem to be experiencing the same thing. I can get the frontend to run, but when I try to do something like create an image, it is unable to contact the backend.

File: devel/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kingsleyzissou) [reply devel/README.md]: Wrong link there ^^ but I added the correct one to the readme :)

**Comment by kingsleyzissou** [comment]
@lucasgarfield when you get a chance can you try verify this? I've added one or two small changes and added the networks back... it seems to be working on my end

**Comment by lavocatt** [comment]
I'm getting this error while following the devel readme:
```

docker-compose build
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 714, in urlopen
    httplib_response = self._make_request(
                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 415, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib64/python3.11/http/client.py", line 1286, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1332, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1281, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1041, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.11/http/client.py", line 979, in send
    self.connect()
  File "/usr/lib/python3.11/site-packages/docker/transport/unixconn.py", line 30, in connect
    sock.connect(self.unix_socket)
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 798, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/util/retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/packages/six.py", line 718, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 714, in urlopen
    httplib_response = self._make_request(
                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 415, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib64/python3.11/http/client.py", line 1286, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1332, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1281, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.11/http/client.py", line 1041, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.11/http/client.py", line 979, in send
    self.connect()
  File "/usr/lib/python3.11/site-packages/docker/transport/unixconn.py", line 30, in connect
    sock.connect(self.unix_socket)
urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/docker/api/client.py", line 214, in _retrieve_server_version
    return self.version(api_version=False)["ApiVersion"]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/daemon.py", line 181, in version
    return self._result(self._get(url), json=True)
                        ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/utils/decorators.py", line 46, in inner
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/client.py", line 237, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/adapters.py", line 547, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/docker-compose", line 33, in <module>
    sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/compose/cli/main.py", line 81, in main
    command_func()
  File "/usr/lib/python3.11/site-packages/compose/cli/main.py", line 200, in perform_command
    project = project_from_options('.', options)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/compose/cli/command.py", line 60, in project_from_options
    return get_project(
           ^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/compose/cli/command.py", line 152, in get_project
    client = get_client(
             ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/compose/cli/docker_client.py", line 41, in get_client
    client = docker_client(
             ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/compose/cli/docker_client.py", line 170, in docker_client
    client = APIClient(use_ssh_client=not use_paramiko_ssh, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/client.py", line 197, in __init__
    self._version = self._retrieve_server_version()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/client.py", line 221, in _retrieve_server_version
    raise DockerException(
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```

**Comment by kingsleyzissou** [comment]
That looks more like a `docker` error than an error with the stack

**Comment by kingsleyzissou** [comment]
So I have a couple of thoughts on this PR and how to improve things. I think ultimately, chatting with @ochosi, we will probably move the docker stuff out of this repo and most likely into the `osbuild/osbuild-getting-started` repo. I just have to make some changes there first.

The second thing I suggest is maybe decoupling the frontend from the stack. I think we needed to have it in the docker stack because of the insights-proxy (I could be wrong though). With the `webpack` changes Jacob made, we could run the frontend separately and point it towards the backend stack.  Thoughts @lucasgarfield & @jkozol (or anyone else)?

**Comment by lavocatt** [comment]
https://github.com/RedHatInsights/image-builder-frontend/pull/1370 is a go, this PR can be closed. Thanks @kingsleyzissou!

**Pull request closed without merging**